woocommerce/woocommerce-gutenberg-products-block
# Product Gallery Block: Add Product Gallery template to allow users to edit full mode view

## What

This PR is the first chunk of work to fix woocommerce/woocommerce-blocks#9942. This PR introduces the Product Gallery template part. In this way, via the Site Editor, the user can customize the full view mode of the Product Gallery.
In this PR, when the Product Gallery full view mode is enabled, the template part is loaded and hidden in the page. When the user clicks on the main image, thanks to the Interactivity API, we open the dialog. The points below will be addressed in other PRs:
- [ ] Improve dialog UI. https://github.com/woocommerce/woocommerce-blocks/pull/10947
- [ ] Create a polished Product Gallery Template part. https://github.com/woocommerce/woocommerce/issues/42217
- [ ] ~~Open the dialog when the user lands on the product page with the query param `?full-screen`~~.
- [ ] Add button on top of the dialog to close it. https://github.com/woocommerce/woocommerce-blocks/pull/10947


Attempt to fix woocommerce/woocommerce-blocks#9942. This is just a POC. 

## Why

Read https://github.com/woocommerce/woocommerce-blocks/discussions/10828.




## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Edit the Single Product Template.
2. Add the Product Gallery block.
3. Ensure that the `Full-screen when clicked` is enabled.
4. Ensure that in the template parts list, the Product Gallery is visible.
5. Edit the Product Gallery template part.
6. Add a paragraph.
7. Save it.
8. Visit a product page.
9. Click on the main image.
10. Ensure that the modal is opened.
11. Ensure that the paragraph is visible


## Screenshots or screencast



https://github.com/woocommerce/woocommerce-blocks/assets/4463174/ce5cac80-2071-4a81-bd5b-9d81a7abb244

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> Add suggested changelog entry here.
